src/main/java/org/wso2/carbon/connector/pojo/ConnectionConfiguration.java

```diff
@@ -0,0 +1,290 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.connector.pojo;
+
+import org.wso2.carbon.connector.connection.EmailProtocol;
+import org.wso2.carbon.connector.exception.InvalidConfigurationException;
+import org.wso2.carbon.connector.utils.EmailConstants;
+
+/**
+ * Configuration parameters used to establish a connection to the email server
+ */
+public class ConnectionConfiguration {
+
+    private String host;
+    private String port;
+    private String connectionName;
+    private String username;
+    private String password;
+    private EmailProtocol protocol;
+    private String readTimeout;
+    private String connectionTimeout;
+    private String writeTimeout;
+    private boolean requireTLS;
+    private boolean checkServerIdentity;
+    private String trustedHosts;
+    private String sslProtocols;
+    private String cipherSuites;
+    private int maxActiveConnections;
+    private int maxIdleConnections;
+    private long maxWaitTime;
+    private long minEvictionTime;
+    private long evictionCheckInterval;
+    private String exhaustedAction;
+    private String initialisationPolicy;
+    private boolean disablePooling;
+
+    public String getHost() {
+
+        return host;
+    }
+
+    public void setHost(String host) throws InvalidConfigurationException {
+        if (host == null){
+            throw new InvalidConfigurationException("Mandatory parameter 'host' is not set.");
+        }
+        this.host = host;
+    }
+
+    public String getPort() {
+
+        return port;
+    }
+
+    public void setPort(String port) throws InvalidConfigurationException {
+        if (port == null){
+            throw new InvalidConfigurationException("Mandatory parameter 'port' is not set.");
+        }
```

Comment:
Shall we check if the passed port resolves to a number?

Reply:
Fixed